pennylane_lightning/src/Apply.cpp

```diff
@@ -90,17 +90,16 @@ void Pennylane::constructAndApplyOperation(
 }
 
 void Pennylane::apply(
-    pybind11::array_t<CplxType>& stateNumpyArray,
-    vector<string> ops,
-    vector<vector<unsigned int>> wires,
-    vector<vector<double>> params,
+    StateVector& state,
+    vector<string>& ops,
+    vector<vector<unsigned int>>& wires,
+    vector<vector<double>>& params,
```

Comment:
None of these are expected to change within `apply` at the moment, so we should be able to define them as `const` (applies in `.hpp` too if accepted).
```suggestion
    const vector<string>& ops,
    const vector<vector<unsigned int>>& wires,
    const vector<vector<double>>& params,
```